### PR TITLE
[WM-2076] Store WDS and Workflows app proxy urls in State

### PR DESCRIPTION
**Pull request by salonishah11**
### What
- Created a new `workflowsAppStore` in state to store WDS, CBAS and Cromwell proxy urls so that it can be reused once loaded across different Azure Workflows components

### Why
- To improve the performance of loading of various components in Azure Workflows tab

### Testing strategy
- Tested locally
- Added related unit tests in SubmissionConfig.test.js

Note: console statements were added only for demo purposes and have been removed

For already existing Workflows App in a workspace:
- When SubmitWorkflow page is loaded for the first time the proxy urls are not set in the `workflowsAppStore`. Hence a call is made to Leonardo to resolve these urls and then set in `workflowsAppStore`
- When user navigates to the SubmissionConfig page, the proxy urls are already loaded in the state and no call is made to Leonardo
- When user navigates back to SubmitWorkflow again, proxy urls are still loaded in the state and this time no call to Leonardo is made

https://github.com/DataBiosphere/terra-ui/assets/16748522/a7436e4c-39ba-4d7e-bc5f-546c98151215

----

For a workspace where the Workflows app is still in provisioning state:
- When user loads the Workflows tab, they will see the spinning icon with text saying that Workflows app is still loading...
- In the background Terra UI will ping Leo to fetch the proxy urls every 30 seconds
- When Workflows app is up and running, the screen will change to list of methods 
-Note: the slight lag when navigating to SubmissionConfig screen at the end was because WDS app was not up at the time the page was loaded but in the next call it was ready

https://github.com/DataBiosphere/terra-ui/assets/16748522/2636bed1-7586-439b-bf6a-f0410d45f158

----

When user switches to Workflows tab for workspace B, after loading Workflows tab of workspace A, call to Leo is made to get apps for workspace B and resolve proxy urls for that workspace 

https://github.com/DataBiosphere/terra-ui/assets/16748522/85e5ce1c-e1bd-4a01-b163-3e3c07169da5

### CODE REVIEW

File: src/libs/state.js [hunk 1]
```diff
@@ -100,3 +100,15 @@ window.ajaxOverridesStore = ajaxOverridesStore;
...
 window.configOverridesStore = configOverridesStore;
+
+/*
+ * Stores the proxy urls for WDS and Azure Workflows apps for a workspace.
+ * Status can be one of None, Ready and Error. The proxy url will be in 'state' field when 'status' is Ready.
+ * When 'state' is Error the 'state' field will contain the error that was returned from Leo (if any).
+ */
+export const workflowsAppStore = Utils.atom({
+  workspaceId: undefined,
+  wdsProxyUrlState: { status: 'None', state: '' },
```
Comment (mspector) [inline src/libs/state.js]: what do you think about using an enum-like object in the `status` field instead of a string?

File: src/libs/utils.ts [hunk 2]
```diff
@@ -460,7 +460,8 @@ export const truncateInteger = (integer) => {
  * Polls using a given function until the pollUntil function returns true.
  * @param pollFn - The function to poll using
  * @param pollTime - How much time there should be in ms between calls of the pollFn
- * @param leading - Whether the function should wait {pollTime} ms before running for the first time
+ * @param leading - If true, the {pollFn} will run once before the next scheduled poll
```
Comment (mspector) [inline src/libs/utils.ts]: I know I suggested this wording, but reading it again, I think it'd be clearer to say:

`If true, the {pollFn} will run once before the first scheduled poll`

**Review by mspector — APPROVED** [review]
I made two minor suggestions that I don't think should block approval

File: src/libs/state.js (hunk 1 above)
Comment (salonishah11) [reply src/libs/state.js]: Yeah that's a good point. The status should be one of 3 values so it makes sense for it to be enum. I will make that change 👍

**Review by dillydally414 — APPROVED** [review]

**Pull request merged**